packages/store/src/StoreCore.sol

```diff
@@ -521,54 +521,31 @@ library StoreCore {
     bytes32 tableId,
     bytes32[] memory keyTuple,
     FieldLayout fieldLayout
-  ) internal view returns (bytes memory) {
+  ) internal view returns (bytes memory staticData, PackedCounter encodedLengths, bytes memory dynamicData) {
```

Comment:
since we're now passing around this tuple a lot, do you think we'd get any value from encoding this as a struct? or would that make it less convenient?

```solidity
struct ValueData {
  bytes staticData;
  PackedCounter encodedLengths;
  bytes dynamicData;
}
```

Reply:
I see a lot of folks asking about stack too deep errors, is this anything to consider?

Reply:
yes, it would reduce vars from 3 to 1

Reply:
might be worth it to support more variables in generated libraries? Since the functions are not used a lot outside codegen I think the slightly decreased dev ex of having to import a struct for the return data is fine

> edit: nevermind, talked myself out of this idea

curious why?

Reply:
the main advantage of using the struct is in the `set` function, where I believe we'd currently run into "stack too deep" issues for tables where num keys + num fields is greater than 7 (because the set function has 5 additional variables besides the key+fields). That might be worked around if we make the version of setter that uses the table's data struct a "first class citizen" instead of internally calling the version with arguments.

In any case, i think we can do this as a follow up since it involves refactors to code unrelated to this PR. Opened an issue to track here: #1535